scvi/module/_amortizedlda.py

```diff
@@ -298,40 +298,51 @@ def topic_by_gene(self, give_mean: bool = True) -> torch.Tensor:
             self.guide.topic_gene_posterior_mu.detach().cpu(),
             self.guide.topic_gene_posterior_sigma.detach().cpu(),
         )
-        if give_mean:
-            return F.softmax(topic_gene_posterior_mu, dim=1)
-        return F.softmax(
-            torch.normal(topic_gene_posterior_mu, topic_gene_posterior_sigma),
-            dim=1,
+        return torch.mean(
+            F.softmax(
+                torch.normal(
+                    topic_gene_posterior_mu.unsqueeze(2).repeat(1, 1, n_samples),
+                    topic_gene_posterior_sigma.unsqueeze(2).repeat(1, 1, n_samples),
+                ),
```

Comment:
we typically use[ torch Normal distribution](https://pytorch.org/docs/stable/distributions.html#normal) object and its sample shape argument